tests/lib/rules/no-this-in-sfc.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @fileoverview Report "this" being used in stateless functional components.
+ */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Constants
+// ------------------------------------------------------------------------------
+
+const ERROR_MESSAGE = 'Stateless functional components should not use this';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-this-in-sfc');
+const RuleTester = require('eslint').RuleTester;
+
+const parserOptions = {
+  ecmaVersion: 8,
+  sourceType: 'module',
+  ecmaFeatures: {
+    experimentalObjectRestSpread: true,
+    jsx: true
+  }
+};
+
+const ruleTester = new RuleTester({parserOptions});
+ruleTester.run('no-this-in-sfc', rule, {
+  valid: [{
+    code: `
+    function Foo(props) {
```

Comment:
Sorry, this is really a nit :) don't these template literals need an extra indentation?

```
code: `
  mystring...
`,
```

Reply:
I think the need is up to personal preference. I can change it if that is the standard preference.

Reply:
imo the indentation of test cases in a non-indentation rule isn't that important, as long as it's readable.